### PR TITLE
first version of CI/CD with GitHub Actions

### PR DESCRIPTION
**Pull request by BuDDi**
- CI / CD for what we need 
- update of the GitHub Actions that have been merged in with Openfire version 4.6.0 integration
- adjusted process to our S3 Maven repository for plugins

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+    steps:
+      # Checks-out the repository under $GITHUB_WORKSPACE, so the job can access it
+      - uses: actions/checkout@v2
+
+      # build docker image
+      - name: Docker build
+        run: |
+          docker build --build-arg KEYSTORE_PWD=${{ secrets.KEYSTORE_PWD }} -t 556016385979.dkr.ecr.eu-central-1.amazonaws.com/feinfone:75 .
```
Comment (nsobadzhiev) [inline .github/workflows/deploy.yml]: it's always building 75?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+      # deploy docker image on AWS
+      - name: AWS ECR deploy
+        # TODO maybe hide some details behind GitHub secrets
+        run: |
+          aws ecr get-login-password --region eu-central-1 | docker login --username AWS --password-stdin 556016385979.dkr.ecr.eu-central-1.amazonaws.com/feinfone
+          docker push 556016385979.dkr.ecr.eu-central-1.amazonaws.com/feinfone:75
```
Comment (nsobadzhiev) [inline .github/workflows/deploy.yml]: Maybe also inject that? It's an AWS account specific URL

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (BuDDi) [reply .github/workflows/deploy.yml]: resolved it hopefully

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (BuDDi) [reply .github/workflows/deploy.yml]: done

File: Dockerfile_maven [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# syntax = docker/dockerfile:1.0-experimental
```
Comment (nsobadzhiev) [inline Dockerfile_maven]: Why do you need another Dockerfile for this (Dockerfile_maven)?
Isn't the idea to replace checking out and building the plugins with just getting them via maven?

**Review by nsobadzhiev — APPROVED** [review]

**Pull request merged**